### ISSUE TITLE
YOLO-Semantic is here: dense pixel-level scene understanding 🌆

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/ultralytics#24845]
YOLO-Semantic brings dense semantic segmentation to Ultralytics YOLO: one class label for every pixel, using the same familiar `ultralytics` Python and CLI workflows.

This is the launch hub for YOLO26 semantic segmentation models, datasets, outputs, and examples.

## What shipped

- **New semantic segmentation task**: `semantic`
- **YOLO26 semantic model family**: `yolo26n-sem.pt`, `yolo26s-sem.pt`, `yolo26m-sem.pt`, `yolo26l-sem.pt`, `yolo26x-sem.pt`
- **Dense class-map outputs** through `result.semantic_mask`
- **Cityscapes-pretrained models** for urban scene understanding
- **Dataset support for PNG masks and YOLO polygon labels**
- **Train, val, predict, and export workflows** through the standard Ultralytics API

## Start here

- Semantic segmentation task docs: https://docs.ultralytics.com/tasks/semantic
- Semantic dataset format: https://docs.ultralytics.com/datasets/semantic
- Cityscapes dataset: https://docs.ultralytics.com/datasets/semantic/cityscapes
- Cityscapes8 quick test dataset: https://docs.ultralytics.com/datasets/semantic/cityscapes8
- Predict mode semantic outputs: https://docs.ultralytics.com/modes/predict#semanticmask
- YOLO26 model docs: https://docs.ultralytics.com/models/yolo26
- Main repository: https://github.com/ultralytics/ultralytics
- Roadmap: https://www.ultralytics.com/roadmap

## Quickstart: CLI

```bash
pip install -U ultralytics

# Predict dense semantic masks
yolo semantic predict model=yolo26n-sem.pt source=https://ultralytics.com/images/bus.jpg imgsz=1024

# Train on the tiny Cityscapes8 smoke-test dataset
yolo semantic train data=cityscapes8.yaml model=yolo26n-sem.pt epochs=10 imgsz=1024

# Validate on Cityscapes-style semantic data
yolo semantic val data=cityscapes8.yaml model=yolo26n-sem.pt imgsz=1024

# Export for deployment
yolo semantic export model=yolo26n-sem.pt format=onnx imgsz=1024
```

## Quickstart: Python

```python
from ultralytics import YOLO

model = YOLO("yolo26n-sem.pt")

results = model.predict("https://ultralytics.com/images/bus.jpg", imgsz=1024)
for result in results:
    semantic = result.semantic_mask
    print(semantic.data.shape)  # H x W dense class-ID map
    print(semantic.names)       # class names
```

## Training from custom data

Semantic datasets can use either mask images or existing YOLO polygon labels.

### Mask-based dataset YAML

```yaml
path: /path/to/my-semantic-dataset
train: images/train
val: images/val
masks_dir: masks
names:
  0: road
  1: sidewalk
  2: building
```

```bash
yolo semantic train data=my-semantic.yaml model=yolo26n-sem.pt epochs=100 imgsz=1024
```

### Polygon-label dataset YAML

If your dataset already uses Ultralytics YOLO polygon labels, omit `masks_dir`. The loader converts polygons into dense semantic masks during training.

```yaml
path: /path/to/my-yolo-polygon-dataset
train: images/train
val: images/val
names:
  0: object
```

## Why it matters

Semantic segmentation is the right tool when the application needs full-scene understanding rather than object instances alone: road scenes, industrial surfaces, medical regions, satellite imagery, robotics, and inspection workflows.

Use YOLO-Semantic when you need dense pixel labels. Use YOLO instance segmentation when you need separate object masks for each object instance.

## Related resources

- Instance segmentation: https://docs.ultralytics.com/tasks/segment
- Detection: https://docs.ultralytics.com/tasks/detect
- Export mode: https://docs.ultralytics.com/modes/export
- Python API: https://docs.ultralytics.com/usage/python
- CLI syntax: https://docs.ultralytics.com/usage/cli
- Minimum reproducible examples: https://docs.ultralytics.com/help/minimum-reproducible-example

Share feedback here for YOLO-Semantic docs, examples, dataset format questions, and workflows that should be easier to discover.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for sharing this Ultralytics update on YOLO26 semantic segmentation 🚀🌆 This is an automated response to help guide contributors and readers, and an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

If this issue leads to any 🐛 bug reports during semantic workflow testing, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it.

If this becomes a custom training ❓ question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments, with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.
